fleetspeak/src/client/generic/proto/fleetspeak_client_generic/config.proto

```diff
@@ -38,6 +38,9 @@ message Config {
 
   // If set, used for SNI and certificate validation.
   string server_name = 9;
+  
+  // If set, the client will prefer HTTP as a Transport
+  bool prefer_http2 = 10;
```

Comment:
I think this setting is more suitable for the CommunicatorConfig (fleetspeak/src/client/proto/fleetspeak_client/client.proto). See https://github.com/google/fleetspeak/pull/518/files for how i have added an option to CommunicatorConfig.

In the streaming communicator code, it can be accessed with `c.conf`. Seems more ergonomic than hijacking the ServerInfo struct in this case.

As a side note, this should have probably also been done for other options. But I admit it's a bit odd that we have a separate config files for the communicator and the client itself in the first place.

Reply:
Thanks heaps Torben for your feedback.
I will adjust accordingly and push the updates shortly.